app/controllers/active_matches_controller.rb

```diff
@@ -18,9 +18,17 @@ def index
     @matches = match_scope
     @current_step = params[:current_step]
     @current_program = params[:current_program]
+    @current_contact_type = params[:current_contact_type]&.to_sym
+    if current_user.can_view_all_matches?
+      @current_filter_contact = params[:current_filter_contact]
+      @current_filter_contact = @current_filter_contact.to_i if @current_filter_contact.present?
+    elsif @current_contact_type.present?
+      @current_filter_contact = current_user.contact&.id
+    end
```

Comment:
Maybe:
```suggestion
    @current_filter_contact = if current_user.can_view_all_matches?
      params[:current_filter_contact].to_i if params[:current_filter_contact].present?
@current_filter_contact.present?
    elsif @current_contact_type.present?
      current_user.contact&.id
    end
```
That _should_ set @current_filter_contact = nil if there's no `params[:current_filter_contact]`